app/controllers/api/suppliers_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Api
+  class SuppliersController < ApiController
+    def locations
+      paginate locations_from_supplier_moves
+    end
+
+  private
+
+    def locations_from_supplier_moves
+      supplier = Supplier.find(params[:supplier_id])
+
+      Location.joins(:moves_from).where(moves: { supplier_id: supplier.id }).distinct
```

Comment:
I like the simplicity of this, but there's a potential issue with pagination as it doesn't specify any ordering - so requesting a few pages in sequence doesn't guarantee the client would get the correct locations in each page. Maybe add an order by clause on something which is unique, perhaps the `nomis_agency_id` on locations?

Also this looks like it has an N+1 query for suppliers; check the development logs when hitting the endpoint via Postman. Would be good to resolve that if possible.

Reply:
We validate the presence of `key`, so we could order by key.
I'll check about the N+1 queries issues, it might be enough just to change joins to includes 🤞

Reply:
I've checked for the N+1 error, and it looks like the ONLY query generated is:
```SELECT DISTINCT "locations".* FROM "locations" INNER JOIN "moves" ON "moves"."from_location_id" = "locations"."id" WHERE "moves"."supplier_id" = $1 ORDER BY "locations"."key" ASC LIMIT $2  [["supplier_id", "xxxxxxx"], ["LIMIT", 11]]```

I've also created more locations and suppliers in my tests, to highlight the issue, but that's the only query I got in the test log.

Reply:
Hmm.. strange. I pulled this branch and ran locally to test the endpoint via Postman and saw extra queries for each supplier. I suspect it's the LocationSerializer that's causing the extra supplier fetches. Have you tried via postman with a bit more data rather than just the test data? I used an old dump from staging. You can resolve it with a simple `includes(:supplier)` anyway.

Reply:
yes, you are right,  I was adding the includes with the moves, and it did not make any difference,
of course we should do it with the suppliers (`.includes(:suppliers)`) 
since you can add them with the `include` params

Fixed!